std/os/bits/wasi.zig

```diff
@@ -9,10 +13,7 @@ pub const ADVICE_WILLNEED: advice_t = 3;
 pub const ADVICE_DONTNEED: advice_t = 4;
 pub const ADVICE_NOREUSE: advice_t = 5;
 
-pub const ciovec_t = extern struct {
-    buf: [*]const u8,
-    buf_len: usize,
-};
+pub const ciovec_t = iovec_const;
```

Comment:
`use @import("../bits.zig");` is needed for this to work

Reply:
Patch to fix WASI:
```
diff --git a/std/os/bits/wasi.zig b/std/os/bits/wasi.zig
index a6f96de3..93d2a82f 100644
--- a/std/os/bits/wasi.zig
+++ b/std/os/bits/wasi.zig
@@ -1,2 +0,0 @@
-use @import("../bits.zig");
-
@@ -15,2 +12,0 @@ pub const ADVICE_NOREUSE: advice_t = 5;
-pub const ciovec_t = iovec_const;
-
@@ -185,2 +180,0 @@ pub const inode_t = u64;
-pub const iovec_t = iovec;
-
diff --git a/std/os/wasi.zig b/std/os/wasi.zig
index 1ce0fe0c..adfe9e82 100644
--- a/std/os/wasi.zig
+++ b/std/os/wasi.zig
@@ -8 +8 @@ pub const is_the_target = builtin.os == .wasi;
-pub use @import("bits/wasi.zig");
+pub use @import("bits.zig");
@@ -20,0 +21,3 @@ comptime {
+pub const iovec_t = iovec;
+pub const ciovec_t = iovec_const;
+
```

Reply:
Ahh that's much better, incorporated, thanks!

Reply:
And here’s for when the docs start to fail:
```patch
diff --git a/doc/langref.html.in b/doc/langref.html.in                          
index 19839904..402e5365 100644                                                 
--- a/doc/langref.html.in                                                       
+++ b/doc/langref.html.in                                                       
@@ -9011,0 +9012 @@ export fn add(a: i32, b: i32) void {                        
+      {#code_release_fast#}                                                    
@@ -9015,2 +9016,2 @@ pub fn main() !void {                                     
-    const args = try std.os.argsAlloc(std.heap.wasm_allocator);                
-    defer std.os.argsFree(std.heap.wasm_allocator, args);                      
+    const args = try std.process.argsAlloc(std.heap.wasm_allocator);           
+    defer std.process.argsFree(std.heap.wasm_allocator, args);
```

Reply:
what's the release fast for?

Reply:
Building WASI in debug mode fails with Unsupported OS in debug.zig. I think this is the correct behavior as we don’t have support for debugging symbols for WASM.

Reply:
Ah. Well that's a separate issue but we can easily make it work, sans debug symbols. And then it can go from "debug mode isn't yet supported" to "debug symbols aren't yet supported", which I think is a preferable state to be in.

Reply:
Agreed. It woks on master. I didn’t want CI to fail due to WAS{M,I}.